pkg/odo/cli/component/create.go

```diff
@@ -656,6 +643,11 @@ func (co *CreateOptions) Validate() (err error) {
 			spinner := log.Spinner("Validating devfile component")
 			defer spinner.End(false)
 
+			// This checks for the presence of s2i style component
+			if util.CheckPathExists(ConfigFilePath) {
+				return errors.New("This directory already contains a component")
```

Comment:
Should we clarify that this is an s2i style component rather than devfile component?

Reply:
Can we move `util.CheckPathExists(ConfigFilePath)` to the top with the one `util.CheckPathExists(EnvFilePath)` below. Then the code block would be:
```
if util.CheckPathExists(ConfigFilePath) || util.CheckPathExists(EnvFilePath) {
    return errors.New("This workspace directory already contains a devfile component")
}
```
It can validate both devfile and s2i components and we don't need to clarify.

```

Reply:
Actually I already did this cleanup in my PR https://github.com/openshift/odo/pull/3118/files as well, I can handle that if you don't mind.

Reply:
ok